### DISCUSSION TITLE
BlueSky Altitude issue

### DISCUSSION THREAD

**Discussion by conjecture23** [TUDelft-CNS-ATM/bluesky#445] [Q&A] [answered]
Hi All,
In my BlueSky simulations, I am trying to get my aircraft to reach a certain altitude at a certain waypoint. However, BlueSky would only start to climb towards that altitude at that waypoint. Below is a sample of my script. Many thanks in advance.

00:00:00.00>TRAIL ON
00:00:00.00>  CRE  WCM6602    b788     2.8368   101.6767 347.0      2900.0    245.0
00:00:00.00>  PAN  WCM6602
00:00:01.00> ADDWPT WCM6602  2.8368       101.6767
00:00:01.00> WCM6602  AT WCM6602001 DO ALT WCM6602    2900.0
00:00:01.00> WCM6602  AT WCM6602001 DO SPD WCM6602    245.0
00:00:02.00> ADDWPT WCM6602  2.8653       101.6665
00:00:02.00> WCM6602  AT WCM6602002 DO ALT WCM6602    4225.0
00:00:02.00> WCM6602  AT WCM6602002 DO SPD WCM6602    247.0
00:00:03.00> ADDWPT WCM6602  2.8840       101.6604
00:00:03.00> WCM6602  AT WCM6602003 DO ALT WCM6602    5075.0
00:00:03.00> WCM6602  AT WCM6602003 DO SPD WCM6602    249.0
00:00:04.00> ADDWPT WCM6602  2.9022       101.6621
00:00:04.00> WCM6602  AT WCM6602004 DO ALT WCM6602    5875.0
00:00:04.00> WCM6602  AT WCM6602004 DO SPD WCM6602    254.0

**Comment by ProfHoekstra** [comment, marked as answer]
Yes, that is indeed what the ALT command does and is meant to do. When you use DO, you prepare a command to be executed when passing that waypoint. VNAV logic is: climb to the next altitude constraint as soon as you can, descend as late as possible. This creates automatically the ToC: Top of Climb and the ToD Top of Descent
If you want VNAV logic, you should enter the altitude as part of the waypoint:

ADDWPT WCM6602 2.8368     101.6767 2900


This will make the aircraft climb as soon as it is allowed by previous altitude constraints according to the Top of Climb logic: climb fast as soon as you can, as would happen in real FMS. If you want to climb gradually, switch off the TOC logic:

WCM6602 TOC OFF

For the descents the top-of-descent logic: descend fast and as late as possible. To make it descend gradually, switch of the Top of Descent logic:

WCM6602 TOD OFF

There are plans to add videos explaining the logic, but you can also find it on the wiki:

https://github.com/TUDelft-CNS-ATM/bluesky/wiki/addwpt

https://github.com/TUDelft-CNS-ATM/bluesky/wiki/Editing-flight-plans

Best regards,
     Jacco Hoekstra


From: conjecture23 ***@***.***>
Sent: dinsdag 10 januari 2023 07:54
To: TUDelft-CNS-ATM/bluesky ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TUDelft-CNS-ATM/bluesky] BlueSky Altitude issue (Discussion #445)


Hi All,
In my BlueSky simulations, I am trying to get my aircraft to reach a certain altitude at a certain waypoint. However, BlueSky would only start to climb towards that altitude at that waypoint. Below is a sample of my script. Many thanks in advance.

00:00:00.00>TRAIL ON
00:00:00.00> CRE WCM6602 b788 2.8368 101.6767 347.0 2900.0    245.0
00:00:00.00> PAN WCM6602
00:00:01.00> ADDWPT WCM6602 2.8368     101.6767
00:00:01.00> WCM6602 AT WCM6602001 DO ALT WCM6602   2900.0
00:00:01.00> WCM6602 AT WCM6602001 DO SPD WCM6602   245.0
00:00:02.00> ADDWPT WCM6602 2.8653     101.6665
00:00:02.00> WCM6602 AT WCM6602002 DO ALT WCM6602   4225.0
00:00:02.00> WCM6602 AT WCM6602002 DO SPD WCM6602   247.0
00:00:03.00> ADDWPT WCM6602 2.8840     101.6604
00:00:03.00> WCM6602 AT WCM6602003 DO ALT WCM6602   5075.0
00:00:03.00> WCM6602 AT WCM6602003 DO SPD WCM6602   249.0
00:00:04.00> ADDWPT WCM6602 2.9022     101.6621
00:00:04.00> WCM6602 AT WCM6602004 DO ALT WCM6602   5875.0
00:00:04.00> WCM6602 AT WCM6602004 DO SPD WCM6602   254.0

**Reply by conjecture23** [reply]
This was exactly what I wanted. Thank you, Prof Hoekstra!
